client/state/stats/chart-tabs/reducer.js

```diff
@@ -26,16 +26,26 @@ export function counts( state = {}, action ) {
 			const records = get( state, `${ action.siteId }.${ action.period }`, [] ).slice();
 			const recordIds = records.map( count => count[ ID ] );
 
+			let areThereChanges = false;
+
 			// Merge existing records with records from API
 			action.data.forEach( recordFromApi => {
 				const index = recordIds.indexOf( recordFromApi[ ID ] );
 				if ( index >= 0 ) {
-					records[ index ] = { ...records[ index ], ...recordFromApi };
+					const newRecords = { ...records[ index ], ...recordFromApi };
+					areThereChanges = areThereChanges || ! isEqual( newRecords, records[ index ] );
+					records[ index ] = newRecords;
```

Comment:
Just a nit: if there are not changes, there's no need to assign to `records[ index ]`. After changing to:
```js
if ( ! isEqual( newRecords, records[ index ] ) {
  areThereChanges = true;
  records[ index ] = newRecords;
}
```
the code "looks similar" to the `else` branch :smile:

Reply:
Good point! We can maintain the existing references if this particular part of the state tree hasn't changed; we don't need to make sure everything is a new reference.